db/install.xml

```diff
@@ -54,19 +54,29 @@
       </KEYS>
     </TABLE>
 
-    <TABLE NAME="qtype_questionpy_pkgversion" COMMENT="Available QuestionPy Package Versions on the Moodle Server">
+    <TABLE NAME="qtype_questionpy_pkgversion" COMMENT="Available package versions on the Moodle Server">
       <FIELDS>
         <FIELD NAME="id" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="true"/>
         <FIELD NAME="packageid" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="false"/>
-        <FIELD NAME="contextid" TYPE="int" LENGTH="10" NOTNULL="true" DEFAULT="0" SEQUENCE="false"/>
         <FIELD NAME="hash" TYPE="char" LENGTH="255" NOTNULL="true" SEQUENCE="false"/>
         <FIELD NAME="version" TYPE="char" LENGTH="255" NOTNULL="true" SEQUENCE="false"/>
+      </FIELDS>
+      <KEYS>
+        <KEY NAME="primary" TYPE="primary" FIELDS="id"/>
+        <KEY NAME="packageid" TYPE="foreign" FIELDS="packageid" REFTABLE="qtype_questionpy_package" REFFIELDS="id"/>
+      </KEYS>
+    </TABLE>
+
+    <TABLE NAME="qtype_questionpy_source" COMMENT="Package version sources.">
+      <FIELDS>
+        <FIELD NAME="id" TYPE="int" LENGTH="10" SEQUENCE="true" NOTNULL="true"/>
+        <FIELD NAME="pkgversionid" TYPE="int" LENGTH="10" SEQUENCE="false" NOTNULL="true"/>
         <FIELD NAME="timecreated" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="false"/>
         <FIELD NAME="userid" TYPE="int" LENGTH="10" NOTNULL="false" SEQUENCE="false"/>
```

Comment:
Ich finde es ein bisschen unsauber hier das userid Feld einfach auf null zu setzen um zu signalisieren, dass die Paketversion vom Server kommt. Mein Vorschlag wäre, das als Boolean-Flag zu speichern entweder in einem qtype_questionpy_source Datensatz oder direkt in der Paketversion.